graphql/ui_plugin_build_baron.go

```diff
@@ -8,17 +8,113 @@ import (
 	"time"
 
 	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/trigger"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/utility"
 	"github.com/mitchellh/mapstructure"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 )
 
 const (
-	jiraSource = "JIRA"
+	jiraSource    = "JIRA"
+	jiraIssueType = "Build Failure"
 )
 
+// bbFileTicket creates a JIRA ticket for a task with the given test failures.
+func BbFileTicket(taskId string) (bool, error) {
+	taskNotFound := false
+	// Find information about the task
+	t, err := task.FindOne(task.ById(taskId))
+	if err != nil {
+		return taskNotFound, err
+
+	}
+	if t == nil {
+		taskNotFound = true
+		return taskNotFound, errors.Wrap(err, fmt.Sprintf("task not found for id %s", taskId))
+	}
+	env := evergreen.GetEnvironment()
+	settings := env.Settings()
+	queue := env.RemoteQueue()
+	buildBaronProjects := BbGetConfig(settings)
+	n, err := makeNotification(settings, buildBaronProjects[t.Project].TicketCreateProject, t)
+	if err != nil {
+		return taskNotFound, err
+	}
+	ts := utility.RoundPartOfMinute(1).Format(units.TSFormat)
+	err = queue.Put(context.TODO(), units.NewEventSendJob(n.ID, ts))
```

Comment:
I would pass along the context from the resolver to here. While it probably won't ever happen, doing so will allow timing out the query to also time out adding this job